client/extensions/woocommerce/state/sites/promotions/helpers.js

```diff
@@ -98,6 +102,9 @@ export function createCouponUpdateFromPromotion( promotion ) {
 	const productIds = ( appliesTo && appliesTo.productIds ) || undefined;
 	const productCategoryIds = ( appliesTo && appliesTo.productCategoryIds ) || undefined;
 
+	// If end date is null, that means it was enabled but not selected, so use today as default.
+	const endDate = null === promotion.endDate ? new Date().toISOString() : promotion.endDate;
```

Comment:
Looks like that answers my question about startDate being able to equal endDate :til: